### PR TITLE
Funding 2021: Hire specutils/specreduce developer

### PR DESCRIPTION
**Pull request by hamogu**
*The Moore budget earmarks 22500 $ specifically for spectroscopy development. I submit this proposal to ensure that there is at least one spectroscopy proposal,and promise to oversee the work to the best of my abilities, but I'm happy to recind it if more detailed proposals are submitted.*

Spectroscopy is a weak spot for astropy, both specutils, and even more specreduce, need love. This proposal is to hire a developer with experience in spectroscopy from the astropy community to drive development forward.

### CODE REVIEW

**Comment by keflavich** [comment]
I'd like to see a more specific plan than 'hire a developer' - can we identify an appropriate candidate during this proposal phase?  But I support the general motion.

**Comment by kelle** [comment]
If someone wants to raise their hand during this proposal phase, that would be amazing! But in case that doesn't happen, it's my understanding that this proposal is to convene a search committee and do an open search.

**Comment by keflavich** [comment]
Got it.  With the plan in those terms, then, I'd like to clarify the work plan: Will the feature list, to be developed, be designed by this person?  Or is someone going to hand the hired person a list of features to implement?   

I'm asking because I think we need this person to develop the plan, not just implement features - as has been noted [elsewhere](https://docs.google.com/document/d/16whMDgDzcOEAHmXvmKIBZUy6fgCvS7cOJ1KkkVxmF7A/edit?usp=sharing), all of the features we need may already exist, but they are not necessarily all packaged in the way we want them.  So I'd like this search to be appropriately targeted - the developer needs to be able to write the documentation, maybe even more than they need to be able to write code.

**Comment by kelle** [comment]
I also want to point out the $22,500 amount is the earmarked amounts in both Years 1 and 2. Since we didn't spend any of this in Year 1, I would suggest that the budget for this idea could range between $22,500 and $45,000.

**Comment by hamogu** [comment]
Just to echo what @kelle said in different words: Ideally, at least on person actually writes a detailed spectroscopy proposals ("Pay myself XXX $ to implement feature A, B, C from the specutils roadmap"). @keflavich If you have time to do work, please write such a project suggestion or if you have people in mind please encourage them to do so themselves!

This is just a "zero-order baseline proposal" to ensure that money is spend on spectroscopy as budgeted in the Moore grant in the absence of any better proposal.

**Comment by keflavich** [comment]
I can continue to be involved, and would be happy to participate in the search committee, but sadly I don't have any candidates in mind.  As member of the spectroscopy coordinating committee, I would expect my role to be both part of the search committee & management of the position, so it's good to formally state that here.

File: finance/proposal-calls/2021-proposal/hire_spectroscopy_developer.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+### Hire specreduce / specutils developer
+
+### Project team
+Moritz Guenther (@hamogu) for this strawmen proposal
```
Comment (eteq) [inline finance/proposal-calls/2021-proposal/hire_spectroscopy_developer.md:4]: I am willing to join as part of the team

Comment (tepickering) [reply finance/proposal-calls/2021-proposal/hire_spectroscopy_developer.md:4]: i am willing as well. i am also involved with `pypeit` development and am keenly interested in getting the various, separately-funded projects aligned and collaborating more closely.

Comment (eteq) [reply finance/proposal-calls/2021-proposal/hire_spectroscopy_developer.md:4]:
```suggestion
Moritz Guenther (@hamogu), Erik Tollerud (@eteq), Tim Pickering (@tepickering) 
```

**Comment by tepickering** [comment]
since my pandemic brain can't read a dang calendar, i posted my PR too late. at @hamogu's behest, here is the project portion of what i put down. it breaks the work that i see needs doing into three categories. hopefully this can help organize efforts going forward, both paid and volunteer.

#### Develop and maintain specreduce package

- Roadmap and issues in place to define what is needed.

- Highest priority is to implement what is required for it to successfully be used for simple cases. Having something working to build on will make it easier to pull in more volunteers/developers.

- Shepherd/review PRs as they come in; monitor GitHub issues & Slack channels

#### Work with other spectroscopy-related packages and initiatives

- Work on data structures for 2D or ND spectra will likely intersect with specutils; visualization needs will likely intersect with jdaviz. It is expected that upstream contributions will be required.

- Continue working with the [PypeIt](https://pypeit.readthedocs.io/en/latest/) project to help bring it more in line with Astropy standards (e.g. the work I have already done to improve its packaging and testing). Identify methods and algorithms used by PypeIt that would be appropriate to port to specreduce.

- Liaise with other spectroscopic data reduction initiatives such as those at Gemini and LCOGT. Work with them to both help focus specreduce development effort to meet their needs, but also fold existing work into specreduce where possible.

#### Documentation and examples for spectroscopic reduction in python

- Collect existing examples/notebooks and host/organize them within specreduce.

- Create and maintain a python/specreduce version of Phil Massey's famous spectroscopy reduction guide for IRAF. Update this document/notebook as specreduce adds features/functionality.

**Comment by jdswinbank** [comment]
Please react to this comment to vote on this proposal (:+1:, :-1:, or no reaction for +0).

File: finance/proposal-calls/2021-proposal/hire_spectroscopy_developer.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+### Project Summary
+*The Moore budget earmarks 22500 $ specifically for spectroscopy development. I submit this proposal to ensure that there is at least one spectroscopy proposal,and promise to oversee the work to the best of my abilities, but I'm happy to recind it if more detailed proposals are submitted.*
+
+Spectroscopy is a weak spot for astropy, both specutils, and even more specreduce, need love. This proposal is to hire a developer with experience in spectroscopy from the astropy community to drive development forward.
+
+### Project / work
+Spectrsocopy has been discussed in previous coordinatation meetings and both packages have maintainers that have listed items for future development in the issue trackers of these projects. I propose to hire a developer with experience in spectroscopy and select the areas from those plans and issues that that developer is most familiar with and can make the most progress on (e.g. if it's someone who has worked on long-slit spectra before, that might be an area to focus onwith IRAF in the past, then re-implementing the most requested [see issue list] IRAF functionality can be a focus).
```
Comment (pllim) [inline finance/proposal-calls/2021-proposal/hire_spectroscopy_developer.md:12]:
```suggestion
Spectroscopy has been discussed in previous coordination meetings and both packages have maintainers that have listed items for future development in the issue trackers of these projects. I propose to hire a developer with experience in spectroscopy and select the areas from those plans and issues that that developer is most familiar with and can make the most progress on (e.g. if it's someone who has worked on long-slit spectra before, that might be an area to focus onwith IRAF in the past, then re-implementing the most requested [see issue list] IRAF functionality can be a focus).
```

**Comment by bsipocz** [comment]
I'm also a 👍 for @tepickering's proposal above as opposed to an open call via AAS (btw, I don't think AAS is the right venue if we aim for global inclusivity) as I feel the latter would very likely lead to the same situation we have been in the past years of coordinating and talking about spectral reduction but not much progress in the code space.

**Comment by hamogu** [comment]
Personally, I fully agree with @tepickering proposal. I think he should be the to-be-hired person named in this proposal and we save the money and effort for a AAS job posting. (I wish he had posted his proposal before the deadline, then I would have retracted this in favor of his...)

**Comment by tepickering** [comment]
yeah, my bad for not paying attention to the calendar. sorry about that....

**Comment by eteq** [comment]
Congratulations @hamogu!  This proposal has the support of the Astropy Community and has been awarded  $18,000  to implement.  This PR will be merged, and you can contact finance@astropy.org or reach out on Slack to arrange financial details (see more details in the email to astropy-dev).

**Comment by jehturner** [comment]
:+1: Will look forward to Gemini being able to co-ordinate with this effort.

**Pull request merged**